GitCommands/Git/GitItemStatus.cs

```diff
@@ -170,9 +177,14 @@ private void SetFlag(bool isSet, Flags flag)
 
         #endregion
 
-        public Task<GitSubmoduleStatus?>? GetSubmoduleStatusAsync()
+        public async Task<GitSubmoduleStatus?> GetSubmoduleStatusAsync()
         {
-            return _submoduleStatus?.JoinAsync();
+            if (_submoduleStatus is null)
+            {
+                return null;
+            }
+
+            return await _submoduleStatus.JoinAsync();
```

Comment:
Here's a bug that this PR fixes. Previously this method would have returned a null _task_ when `_submoduleStatus` was null. Trying to await a null task results in an exception. The consuming code expected a task containing a null _value_ instead, which the new code now provides.